packages/graph/hash_graph/lib/graph/src/api/rest/link_type.rs

```diff
@@ -35,9 +36,18 @@ use crate::{
 )]
 pub struct LinkTypeResource;
 
+/// Specifies the requirements to a [`Store`] for the link-type REST API.
+///
+/// [`Store`]: crate::store::Store
+pub trait LinkTypeBackend = crud::Read<VersionId, LinkType, Output = Persisted<LinkType>>;
```

Comment:
I do like that we can specify these traits as a requirement as we do here. I have a slight feeling that these constraints on what we need to have implemented are defined a little "far up" in the application stack. These trait "usages" seem like they make sense to be given closer to the "core" of the application.

I feel this way because currently the REST API would impose constraints on how the internal core is implemented (correct me if I'm wrong), and it seems that it would be good to encode the constraints closer to the `Store` :)

Reply:
I don't really mind either way. I like them here as here the functionality is used. The REST API does not care about any implementations on `Store`, as long these bounds are fulfilled. When specifying on `Store`, this would imply, that any `Store`, regardless of where it's used, needs to implement every functionality.

Reply:
We agreed off-GitHub, that these bounds will be moved to the `Store` definition in a follow-up PR, when we split the trait into separated stores (e.g. `DataTypeStore`)